fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/services/CreateResourceServiceImpl.java

```diff
@@ -71,12 +76,17 @@
     private UniqueValueSupplier minter;
 
     @Override
-    public void perform(final String txId, final String fedoraId, final String slug, final String contentType,
-                        final List<String> linkHeaders, final Collection<String> digest,
+    public void perform(final String txId, final String fedoraId, final String slug, final boolean isContained,
+                        final String contentType, final List<String> linkHeaders, final Collection<String> digest,
                         final InputStream requestBody, final ExternalContent externalContent) {
         final PersistentStorageSession pSession = this.psManager.getSession(txId);
-        final String fullPath = doInternalPerform(pSession, fedoraId, slug, linkHeaders);
-        final Collection<URI> uriDigests = digest.stream().map(URI::create).collect(Collectors.toCollection(HashSet::new));
+        checkAclLinkHeader(linkHeaders);
+        checkParent(pSession, (isContained ? fedoraId : findExistingAncestor(fedoraId)));
+
+        final String fullPath = isContained ? getResourcePath(pSession, fedoraId, slug) : fedoraId;
+
+        final Collection<URI> uriDigests = (digest == null ?
```

Comment:
this conditional is backwards

Reply:
Also, you can use `Collections.emptySet()` instead of instantiating a new hashset in the null case, unless we are expecting to modify the set of digests (its possible we might need to add in the system generated checksum later, but we may still not want to change the user supplied set of digests)

Reply:
fixing in next commit